src/zephyr_coap_req.c

```diff
@@ -146,7 +146,9 @@ static int golioth_coap_req_reply_handler(struct golioth_coap_req *req,
     int block2;
     int err;
 
-    code = coap_header_get_code(response);
+    // TODO: Replace with coap_header_get_code(response) after
+    // NCS update includes fix for TOO_MANY_REQUESTS
+    code = response->data[1];
```

Comment:
Is `response` and the data member being checked for null somewhere? This is present in the Zephyr function: https://github.com/zephyrproject-rtos/zephyr/blob/8f4ac0d4ab450d17c852a5af2c80a8f6500f985e/subsys/net/lib/coap/coap.c#L958

Reply:
Yeah, there are several calls into the CoAP stack before `coap_header_get_code()` that also check for NULL and would have returned an error before this point.